ddtrace/internal/datadog/profiling/dd_wrapper/src/profile.cpp

```diff
@@ -196,11 +197,20 @@ Datadog::Profile::val()
 }
 
 bool
-Datadog::Profile::collect(const ddog_prof_Sample& sample, int64_t endtime_ns)
+Datadog::Profile::collect(const ddog_prof_Sample2& sample, int64_t endtime_ns)
 {
     static bool already_warned = false; // cppcheck-suppress threadsafety-threadsafety
     const std::lock_guard<std::mutex> lock(profile_mtx);
-    auto res = ddog_prof_Profile_add(&cur_profile, sample, endtime_ns);
+
+    auto set_dictionary_res = ddog_prof_Profile_set_profiles_dictionary(&cur_profile, get_dictionary());
```

Comment:
The `ddog_prof_Profile_set_profiles_dictionary` function is going to reset some internal tables every time this is called. This should be moved to wherever you create a new profile.

```rs
    /// Sets the profiles dictionary to enable `api2` operations.
    pub fn set_profiles_dictionary(
        &mut self,
        profiles_dictionary: crate::profiles::collections::Arc<ProfilesDictionary>,
    ) {
        self.profiles_dictionary_translator = Some(ProfilesDictionaryTranslator {
            profiles_dictionary,
            mappings: Default::default(),
            functions: Default::default(),
            strings: Default::default(),
        });
    }
```